tests/v2/validation/observability/stackstate_test.go

```diff
@@ -0,0 +1,271 @@
+	require.NoError(ss.T(), err)
+
+	log.Info("Verifying if the ui plugin repository for ui extensions exists.")
+	_, err = ss.client.Catalog.ClusterRepos().Get(context.TODO(), rancherUIPlugins, meta.GetOptions{})
+
+	if k8sErrors.IsNotFound(err) {
+
+		err = observability.AddExtensionsRepo(ss.client, rancherUIPlugins, uiExtensionsRepo, uiGitBranch)
+		require.NoError(ss.T(), err)
+	} else {
+		require.NoError(ss.T(), err)
+	}
+
+	var stackstateConfigs observability.StackStateConfigs
+	config.LoadConfig(stackStateConfigFileKey, &stackstateConfigs)
+	ss.stackstateConfigs = stackstateConfigs
+
+	log.Info("Crete a node driver with stackstate extensions ui to whitelist stackstate URL")
+	err = observability.InstallNodeDriver(ss.client, []string{ss.stackstateConfigs.Url})
+	require.NoError(ss.T(), err)
+
+	ss.T().Log("Checking if the stack state CRD is installed.")
+	crdsExists, err := ss.client.Steve.SteveType(observability.ApiExtenisonsCRD).ByID(observability.ObservabilitySteveType)
+	if crdsExists == nil && strings.Contains(err.Error(), "Not Found") {
+		err = observability.InstallStackstateCRD(ss.client)
+		require.NoError(ss.T(), err)
+	} else {
+		require.NoError(ss.T(), err)
+	}
+
+	client, err = client.ReLogin()
+	require.NoError(ss.T(), err)
+
+	ss.T().Log("Checking if the stack state extension is already installed.")
+	initialStackstateExtension, err := extencharts.GetChartStatus(client, localCluster, charts.StackstateExtensionNamespace, charts.StackstateExtensionsName)
+	require.NoError(ss.T(), err)
+
+	if !initialStackstateExtension.IsAlreadyInstalled {
+		latestUIPluginVersion, err := ss.client.Catalog.GetLatestChartVersion(charts.StackstateExtensionsName, charts.UIPluginName)
+		require.NoError(ss.T(), err)
+
+		ss.T().Log("Installing stackstate ui extensions")
+		extensionOptions := &charts.ExtensionOptions{
+			ChartName:   charts.StackstateExtensionsName,
+			ReleaseName: charts.StackstateExtensionsName,
+			Version:     latestUIPluginVersion,
+		}
+
+		err = charts.InstallStackstateExtension(client, extensionOptions)
+		require.NoError(ss.T(), err)
+
+		log.Info("Adding stack state extension configuration.")
+
+		steveAdminClient, err := client.Steve.ProxyDownstream(localCluster)
+		require.NoError(ss.T(), err)
+
+		crdConfig := observability.NewStackstateCRDConfiguration(charts.StackstateNamespace, ss.stackstateConfigs)
+		crd, err := steveAdminClient.SteveType(charts.StackstateCRD).Create(crdConfig)
+		require.NoError(ss.T(), err)
+
+		_, err = steveAdminClient.SteveType(charts.StackstateCRD).ByID(crd.ID)
+		require.NoError(ss.T(), err)
+	}
+
+	latestSSVersion, err := ss.client.Catalog.GetLatestChartVersion(charts.StackstateK8sAgent, rancherPartnerCharts)
+	ss.stackstateAgentInstallOptions = &charts.InstallOptions{
+		Cluster:   cluster,
+		Version:   latestSSVersion,
+		ProjectID: project.ID,
+	}
+}
+
+func (ss *StackStateTestSuite) TestStackState() {
```

Comment:
can you rename this to something more descriptive. Remember, `stackstate` is already in ths suite name.